pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -57,9 +58,18 @@ const (
 	ICMPv6DstAdminProhibitedCode uint8 = 1
 )
 
-var (
-	AntreaPolicyLogger *log.Logger
-)
+//AntreaPolicyLogger is used for antrea policy audit logging.
+// Includes a lumberjack logger and a map used for log deduplication.
+type AntreaPolicyLogger struct {
+	anpLogger        PolicyCustomLogger
+	logDeduplication logRecordDedupMap
+}
+
+// PolicyCustomLogger is the base logger interface.
+// log.Logger implements PolicyCustomLogger.
+type PolicyCustomLogger interface {
```

Comment:
@antoninbas I feel like this is a workaround just to make UT easier (i.e. in UT we can use a custom logger to do `Printf` and check if the result matches expectation, without having to write to some file), but I don't have a better idea I guess

Reply:
what I had in mind was providing a custom object that implements the `io.Writer` interface for tests (instead of `logOutput := &lumberjack.Logger{...`) as it avoids another level of interface. The advantage is that Go has a standard package for testing Writers which makes writing test cases easy: https://golang.org/pkg/testing/iotest/